subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -962,4 +962,100 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec {
         """
     }
 
+    def "implementation of nested closure in decorated bean is tracked"() {
+        taskWithNestedDecoratedBean()
+        buildFile << """
+            extensions.create("bean", NestedBean.class)
+            
+            bean {
+                withAction { it.text = "hello" }
+            }
+            
+            task myTask(type: TaskWithNestedDecoratedAction) {
+                bean = project.bean
+            }
+        """
+
+        buildFile.makeOlder()
```

Comment:
I thought I needed this to prevent a side effect in the up-to-date check based on some comments I saw elsewhere, but I realize that scenario doesn't fit this case.  It's not needed and has been removed.